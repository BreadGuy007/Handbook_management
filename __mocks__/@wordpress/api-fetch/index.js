const apiFetch = jest.fn( () => {
	return apiFetch.mockReturnValue;
} );
apiFetch.mockReturnValue = 'Mock this value by overriding apiFetch.mockReturnValue.';

export default apiFetch;
