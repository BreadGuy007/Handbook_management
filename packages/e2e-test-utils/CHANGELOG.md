## 1.0.0 (2019-03-06)

-   Initial release.
