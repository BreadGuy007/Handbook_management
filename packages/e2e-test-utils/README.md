# E2E Test Utils

End-To-End (E2E) test utils for WordPress.

## Installation

Install the module

```bash
npm install @wordpress/e2e-test-utils --save-dev
```

## API

<!-- START TOKEN(Autogenerated API docs) -->

<a name="activatePlugin" href="#activatePlugin">#</a> **activatePlugin** [\<>](src/index.js#L1-L1)

Activates an installed plugin.

_Parameters_

-   _slug_ `string`: Plugin slug.

<a name="arePrePublishChecksEnabled" href="#arePrePublishChecksEnabled">#</a> **arePrePublishChecksEnabled** [\<>](src/index.js#L2-L2)

Verifies if publish checks are enabled.

_Returns_

-   `boolean`: Boolean which represents the state of prepublish checks.

<a name="clearLocalStorage" href="#clearLocalStorage">#</a> **clearLocalStorage** [\<>](src/index.js#L3-L3)

Clears the local storage.

<a name="clickBlockAppender" href="#clickBlockAppender">#</a> **clickBlockAppender** [\<>](src/index.js#L4-L4)

Clicks the default block appender.

<a name="clickBlockToolbarButton" href="#clickBlockToolbarButton">#</a> **clickBlockToolbarButton** [\<>](src/index.js#L5-L5)

Clicks a block toolbar button.

_Parameters_

-   _buttonAriaLabel_ `string`: The aria label of the button to click.

<a name="clickButton" href="#clickButton">#</a> **clickButton** [\<>](src/index.js#L6-L6)

Clicks a button based on the text on the button.

_Parameters_

-   _buttonText_ `string`: The text that appears on the button to click.

<a name="clickOnCloseModalButton" href="#clickOnCloseModalButton">#</a> **clickOnCloseModalButton** [\<>](src/index.js#L7-L7)

Click on the close button of an open modal.

_Parameters_

-   _modalClassName_ `?string`: Class name for the modal to close

<a name="clickOnMoreMenuItem" href="#clickOnMoreMenuItem">#</a> **clickOnMoreMenuItem** [\<>](src/index.js#L8-L8)

Clicks on More Menu item, searches for the button with the text provided and clicks it.

_Parameters_

-   _buttonLabel_ `string`: The label to search the button for.

<a name="createEmbeddingMatcher" href="#createEmbeddingMatcher">#</a> **createEmbeddingMatcher** [\<>](src/index.js#L52-L52)

Creates a function to determine if a request is embedding a certain URL.

_Parameters_

-   _url_ `string`: The URL to check against a request.

_Returns_

-   `function`: Function that determines if a request is for the embed API, embedding a specific URL.

<a name="createJSONResponse" href="#createJSONResponse">#</a> **createJSONResponse** [\<>](src/index.js#L52-L52)

Respond to a request with a JSON response.

_Parameters_

-   _mockResponse_ `string`: The mock object to wrap in a JSON response.

_Returns_

-   `Promise`: Promise that responds to a request with the mock JSON response.

<a name="createNewPost" href="#createNewPost">#</a> **createNewPost** [\<>](src/index.js#L9-L9)

Creates new post.

_Parameters_

-   _obj_ `Object`: Object to create new post, along with tips enabling option.

<a name="createURL" href="#createURL">#</a> **createURL** [\<>](src/index.js#L10-L10)

Creates new URL by parsing base URL, WPPath and query string.

_Parameters_

-   _WPPath_ `string`: String to be serialized as pathname.
-   _query_ `?string`: String to be serialized as query portion of URL.

_Returns_

-   `string`: String which represents full URL.

<a name="createURLMatcher" href="#createURLMatcher">#</a> **createURLMatcher** [\<>](src/index.js#L52-L52)

Creates a function to determine if a request is calling a URL with the substring present.

_Parameters_

-   _substring_ `string`: The substring to check for.

_Returns_

-   `function`: Function that determines if a request's URL contains substring.

<a name="deactivatePlugin" href="#deactivatePlugin">#</a> **deactivatePlugin** [\<>](src/index.js#L11-L11)

Deactivates an active plugin.

_Parameters_

-   _slug_ `string`: Plugin slug.

<a name="disablePrePublishChecks" href="#disablePrePublishChecks">#</a> **disablePrePublishChecks** [\<>](src/index.js#L12-L12)

Disables Pre-publish checks.

<a name="enablePageDialogAccept" href="#enablePageDialogAccept">#</a> **enablePageDialogAccept** [\<>](src/index.js#L13-L13)

Enables even listener which accepts a page dialog which
may appear when navigating away from Gutenberg.

<a name="enablePrePublishChecks" href="#enablePrePublishChecks">#</a> **enablePrePublishChecks** [\<>](src/index.js#L14-L14)

Enables Pre-publish checks.

<a name="ensureSidebarOpened" href="#ensureSidebarOpened">#</a> **ensureSidebarOpened** [\<>](src/index.js#L15-L15)

Verifies that the edit post sidebar is opened, and if it is not, opens it.

_Returns_

-   `Promise`: Promise resolving once the edit post sidebar is opened.

<a name="findSidebarPanelToggleButtonWithTitle" href="#findSidebarPanelToggleButtonWithTitle">#</a> **findSidebarPanelToggleButtonWithTitle** [\<>](src/index.js#L16-L16)

Finds a sidebar panel with the provided title.

_Parameters_

-   _panelTitle_ `string`: The name of sidebar panel.

_Returns_

-   `?ElementHandle`: Object that represents an in-page DOM element.

<a name="findSidebarPanelWithTitle" href="#findSidebarPanelWithTitle">#</a> **findSidebarPanelWithTitle** [\<>](src/index.js#L17-L17)

Finds the button responsible for toggling the sidebar panel with the provided title.

_Parameters_

-   _panelTitle_ `string`: The name of sidebar panel.

_Returns_

-   `?ElementHandle`: Object that represents an in-page DOM element.

<a name="getAllBlockInserterItemTitles" href="#getAllBlockInserterItemTitles">#</a> **getAllBlockInserterItemTitles** [\<>](src/index.js#L18-L18)

Returns an array of strings with all inserter item titles.

_Returns_

-   `Promise`: Promise resolving with an array containing all inserter item titles.

<a name="getAllBlocks" href="#getAllBlocks">#</a> **getAllBlocks** [\<>](src/index.js#L19-L19)

Returns an array with all blocks; Equivalent to calling wp.data.select( 'core/editor' ).getBlocks();

_Returns_

-   `Promise`: Promise resolving with an array containing all blocks in the document.

<a name="getAvailableBlockTransforms" href="#getAvailableBlockTransforms">#</a> **getAvailableBlockTransforms** [\<>](src/index.js#L20-L20)

Returns an array of strings with all block titles,
that the current selected block can be transformed into.

_Returns_

-   `Promise`: Promise resolving with an array containing all possible block transforms

<a name="getBlockSetting" href="#getBlockSetting">#</a> **getBlockSetting** [\<>](src/index.js#L21-L21)

Returns a string containing the block title associated with the provided block name.

_Parameters_

-   _blockName_ `string`: Block name.
-   _setting_ `string`: Block setting e.g: title, attributes....

_Returns_

-   `Promise`: Promise resolving with a string containing the block title.

<a name="getEditedPostContent" href="#getEditedPostContent">#</a> **getEditedPostContent** [\<>](src/index.js#L22-L22)

Returns a promise which resolves with the edited post content (HTML string).

_Returns_

-   `Promise`: Promise resolving with post content markup.

<a name="hasBlockSwitcher" href="#hasBlockSwitcher">#</a> **hasBlockSwitcher** [\<>](src/index.js#L23-L23)

Returns a boolean indicating if the current selected block has a block switcher or not.

_Returns_

-   `Promise`: Promise resolving with a boolean.

<a name="insertBlock" href="#insertBlock">#</a> **insertBlock** [\<>](src/index.js#L24-L24)

Opens the inserter, searches for the given term, then selects the first
result that appears.

_Parameters_

-   _searchTerm_ `string`: The text to search the inserter for.
-   _panelName_ `string`: The inserter panel to open (if it's closed by default).

<a name="installPlugin" href="#installPlugin">#</a> **installPlugin** [\<>](src/index.js#L25-L25)

Installs a plugin from the WP.org repository.

_Parameters_

-   _slug_ `string`: Plugin slug.
-   _searchTerm_ `?string`: If the plugin is not findable by its slug use an alternative term to search.

<a name="isCurrentURL" href="#isCurrentURL">#</a> **isCurrentURL** [\<>](src/index.js#L26-L26)

Checks if current URL is a WordPress path.

_Parameters_

-   _WPPath_ `string`: String to be serialized as pathname.
-   _query_ `?string`: String to be serialized as query portion of URL.

_Returns_

-   `boolean`: Boolean represents whether current URL is or not a WordPress path.

<a name="isInDefaultBlock" href="#isInDefaultBlock">#</a> **isInDefaultBlock** [\<>](src/index.js#L27-L27)

Checks if the block that is focused is the default block.

_Returns_

-   `Promise`: Promise resolving with a boolean indicating if the focused block is the default block.

<a name="loginUser" href="#loginUser">#</a> **loginUser** [\<>](src/index.js#L28-L28)

Performs log in with specified username and password.

_Parameters_

-   _username_ `?string`: String to be used as user credential.
-   _password_ `?string`: String to be used as user credential.

<a name="mockOrTransform" href="#mockOrTransform">#</a> **mockOrTransform** [\<>](src/index.js#L52-L52)

Mocks a request with the supplied mock object, or allows it to run with an optional transform, based on the
deserialised JSON response for the request.

_Parameters_

-   _mockCheck_ `function`: function that returns true if the request should be mocked.
-   _mock_ `Object`: A mock object to wrap in a JSON response, if the request should be mocked.
-   _responseObjectTransform_ `(function|undefined)`: An optional function that transforms the response's object before the response is used.

_Returns_

-   `Promise`: Promise that uses `mockCheck` to see if a request should be mocked with `mock`, and optionally transforms the response with `responseObjectTransform`.

<a name="observeFocusLoss" href="#observeFocusLoss">#</a> **observeFocusLoss** [\<>](src/index.js#L29-L29)

Binds to the document on page load which throws an error if a `focusout`
event occurs without a related target (i.e. focus loss).

<a name="openAllBlockInserterCategories" href="#openAllBlockInserterCategories">#</a> **openAllBlockInserterCategories** [\<>](src/index.js#L30-L30)

Opens all block inserter categories.

<a name="openDocumentSettingsSidebar" href="#openDocumentSettingsSidebar">#</a> **openDocumentSettingsSidebar** [\<>](src/index.js#L31-L31)

Clicks on the button in the header which opens Document Settings sidebar when it is closed.

<a name="openGlobalBlockInserter" href="#openGlobalBlockInserter">#</a> **openGlobalBlockInserter** [\<>](src/index.js#L32-L32)

Opens the global block inserter.

<a name="openPublishPanel" href="#openPublishPanel">#</a> **openPublishPanel** [\<>](src/index.js#L33-L33)

Opens the publish panel.

<a name="pressKeyTimes" href="#pressKeyTimes">#</a> **pressKeyTimes** [\<>](src/index.js#L34-L34)

Presses the given keyboard key a number of times in sequence.

_Parameters_

-   _key_ `string`: Key to press.
-   _count_ `number`: Number of times to press.

_Returns_

-   `Promise`: Promise resolving when key presses complete.

<a name="pressKeyWithModifier" href="#pressKeyWithModifier">#</a> **pressKeyWithModifier** [\<>](src/index.js#L35-L35)

Performs a key press with modifier (Shift, Control, Meta, Alt), where each modifier
is normalized to platform-specific modifier.

_Parameters_

-   _modifier_ `string`: Modifier key.
-   _key_ `string`: Key to press while modifier held.

<a name="publishPost" href="#publishPost">#</a> **publishPost** [\<>](src/index.js#L36-L36)

Publishes the post, resolving once the request is complete (once a notice
is displayed).

_Returns_

-   `Promise`: Promise resolving when publish is complete.

<a name="publishPostWithPrePublishChecksDisabled" href="#publishPostWithPrePublishChecksDisabled">#</a> **publishPostWithPrePublishChecksDisabled** [\<>](src/index.js#L37-L37)

Publishes the post without the pre-publish checks,
resolving once the request is complete (once a notice is displayed).

_Returns_

-   `Promise`: Promise resolving when publish is complete.

<a name="saveDraft" href="#saveDraft">#</a> **saveDraft** [\<>](src/index.js#L38-L38)

Saves the post as a draft, resolving once the request is complete (once the
"Saved" indicator is displayed).

_Returns_

-   `Promise`: Promise resolving when draft save is complete.

<a name="searchForBlock" href="#searchForBlock">#</a> **searchForBlock** [\<>](src/index.js#L39-L39)

Search for block in the global inserter

_Parameters_

-   _searchTerm_ `string`: The text to search the inserter for.

<a name="selectBlockByClientId" href="#selectBlockByClientId">#</a> **selectBlockByClientId** [\<>](src/index.js#L40-L40)

Given the clientId of a block, selects the block on the editor.

_Parameters_

-   _clientId_ `string`: Identified of the block.

<a name="setBrowserViewport" href="#setBrowserViewport">#</a> **setBrowserViewport** [\<>](src/index.js#L41-L41)

Sets browser viewport to specified type.

_Parameters_

-   _type_ `string`: String to represent dimensions type; can be either small or large.

<a name="setPostContent" href="#setPostContent">#</a> **setPostContent** [\<>](src/index.js#L42-L42)

Sets code editor content

_Parameters_

-   _content_ `string`: New code editor content.

_Returns_

-   `Promise`: Promise resolving with an array containing all blocks in the document.

<a name="setUpResponseMocking" href="#setUpResponseMocking">#</a> **setUpResponseMocking** [\<>](src/index.js#L52-L52)

Sets up mock checks and responses. Accepts a list of mock settings with the following properties:

-   `match`: function to check if a request should be mocked.
-   `onRequestMatch`: async function to respond to the request.

_Usage_

```js
const MOCK_RESPONSES = [
  {
    match: isEmbedding( 'https://wordpress.org/gutenberg/handbook/' ),
    onRequestMatch: JSONResponse( MOCK_BAD_WORDPRESS_RESPONSE ),
  },
  {
    match: isEmbedding( 'https://wordpress.org/gutenberg/handbook/block-api/attributes/' ),
    onRequestMatch: JSONResponse( MOCK_EMBED_WORDPRESS_SUCCESS_RESPONSE ),
  }
];
setUpResponseMocking( MOCK_RESPONSES );
```

If none of the mock settings match the request, the request is allowed to continue.

_Parameters_

-   _mocks_ `Array`: Array of mock settings.

<a name="switchEditorModeTo" href="#switchEditorModeTo">#</a> **switchEditorModeTo** [\<>](src/index.js#L43-L43)

Switches editor mode.

_Parameters_

-   _mode_ `string`: String editor mode.

<a name="switchUserToAdmin" href="#switchUserToAdmin">#</a> **switchUserToAdmin** [\<>](src/index.js#L44-L44)

Switches the current user to the admin user (if the user
running the test is not already the admin user).

<a name="switchUserToTest" href="#switchUserToTest">#</a> **switchUserToTest** [\<>](src/index.js#L45-L45)

Switches the current user to whichever user we should be
running the tests as (if we're not already that user).

<a name="toggleScreenOption" href="#toggleScreenOption">#</a> **toggleScreenOption** [\<>](src/index.js#L46-L46)

Toggles the screen option with the given label.

_Parameters_

-   _label_ `string`: The label of the screen option, e.g. 'Show Tips'.
-   _shouldBeChecked_ `?boolean`: If true, turns the option on. If false, off. If undefined, the option will be toggled.

<a name="transformBlockTo" href="#transformBlockTo">#</a> **transformBlockTo** [\<>](src/index.js#L47-L47)

Converts editor's block type.

_Parameters_

-   _name_ `string`: Block name.

<a name="uninstallPlugin" href="#uninstallPlugin">#</a> **uninstallPlugin** [\<>](src/index.js#L48-L48)

Uninstalls a plugin.

_Parameters_

-   _slug_ `string`: Plugin slug.

<a name="visitAdminPage" href="#visitAdminPage">#</a> **visitAdminPage** [\<>](src/index.js#L49-L49)

Visits admin page; if user is not logged in then it logging in it first, then visits admin page.

_Parameters_

-   _adminPath_ `string`: String to be serialized as pathname.
-   _query_ `string`: String to be serialized as query portion of URL.

<a name="waitForWindowDimensions" href="#waitForWindowDimensions">#</a> **waitForWindowDimensions** [\<>](src/index.js#L50-L50)

Function that waits until the page viewport has the required dimensions.
It is being used to address a problem where after using setViewport the execution may continue,
without the new dimensions being applied.
<https://github.com/GoogleChrome/puppeteer/issues/1751>

_Parameters_

-   _width_ `number`: Width of the window.
-   _height_ `height`: Height of the window.


<!-- END TOKEN(Autogenerated API docs) -->

<br/><br/><p align="center"><img src="https://s.w.org/style/images/codeispoetry.png?1" alt="Code is Poetry." /></p>
