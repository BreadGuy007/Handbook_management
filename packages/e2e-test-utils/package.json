{
	"name": "@wordpress/e2e-test-utils",
	"version": "1.0.0-alpha.0",
	"description": "End-To-End (E2E) test utils for WordPress.",
	"author": "The WordPress Contributors",
	"license": "GPL-2.0-or-later",
	"keywords": [
		"wordpress",
		"e2e",
		"utils"
	],
	"homepage": "https://github.com/WordPress/gutenberg/tree/master/packages/e2e-test-utils/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/WordPress/gutenberg.git"
	},
	"bugs": {
		"url": "https://github.com/WordPress/gutenberg/issues"
	},
	"files": [
		"build",
		"build-module"
	],
	"main": "build/index.js",
	"module": "build-module/index.js",
	"dependencies": {
		"@wordpress/keycodes": "file:../keycodes",
		"@wordpress/url": "file:../url",
		"lodash": "^4.17.10",
		"node-fetch": "^1.7.3"
	},
	"peerDependencies": {
		"jest": ">=23",
		"puppeteer": ">=1.6"
	},
	"publishConfig": {
		"access": "public"
	}
}
