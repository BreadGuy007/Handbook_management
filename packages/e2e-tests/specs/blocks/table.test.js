/**
 * WordPress dependencies
 */
import { createNewPost, insertBlock, getEditedPostContent } from '@wordpress/e2e-test-utils';

describe( 'Table', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'displays a form for choosing the row and column count of the table', async () => {
		await insertBlock( 'Table' );

		// Check for existence of the column count field.
		const columnCountLabel = await page.$x( "//div[@data-type='core/table']//label[text()='Column Count']" );
		expect( columnCountLabel ).toHaveLength( 1 );

		// Modify the column count.
		await columnCountLabel[ 0 ].click();
		const currentColumnCount = await page.evaluate( () => document.activeElement.value );
		expect( currentColumnCount ).toBe( '2' );
		await page.keyboard.press( 'Backspace' );
		await page.keyboard.type( '5' );

		// // Check for existence of the row count field.
		const rowCountLabel = await page.$x( "//div[@data-type='core/table']//label[text()='Row Count']" );
		expect( rowCountLabel ).toHaveLength( 1 );

		// // Modify the row count.
		await rowCountLabel[ 0 ].click();
		const currentRowCount = await page.evaluate( () => document.activeElement.value );
		expect( currentRowCount ).toBe( '2' );
		await page.keyboard.press( 'Backspace' );
		await page.keyboard.type( '10' );

		// // Create the table.
		const createButton = await page.$x( "//div[@data-type='core/table']//button[text()='Create']" );
		await createButton[ 0 ].click();

		// // Expect the post content to have a correctly sized table.
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'allows text to by typed into cells', async () => {
		await insertBlock( 'Table' );

		// Create the table.
		const createButton = await page.$x( "//div[@data-type='core/table']//button[text()='Create']" );
		await createButton[ 0 ].click();

		// Click the first cell and add some text.
		await page.click( '.wp-block-table__cell-content' );
		await page.keyboard.type( 'This' );

		// Tab to the next cell and add some text.
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'is' );

		// Tab to the next cell and add some text.
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'table' );

		// Tab to the next cell and add some text.
		await page.keyboard.press( 'Tab' );
		await page.keyboard.type( 'block' );

		// Expect the post to have the correct written content inside the table.
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'allows header and footer rows to be switched on and off', async () => {
		await insertBlock( 'Table' );

		const headerSwitchSelector = "//label[text()='Header section']";
		const footerSwitchSelector = "//label[text()='Footer section']";

		// Expect the header and footer switches not to be present before the table has been created.
		let headerSwitch = await page.$x( headerSwitchSelector );
		let footerSwitch = await page.$x( footerSwitchSelector );
		expect( headerSwitch ).toHaveLength( 0 );
		expect( footerSwitch ).toHaveLength( 0 );

		// Create the table.
		const createButton = await page.$x( "//div[@data-type='core/table']//button[text()='Create']" );
		await createButton[ 0 ].click();

		// Expect the header and footer switches to be present now that the table has been created.
		headerSwitch = await page.$x( headerSwitchSelector );
		footerSwitch = await page.$x( footerSwitchSelector );
		expect( headerSwitch ).toHaveLength( 1 );
		expect( footerSwitch ).toHaveLength( 1 );

		// Toggle on the switches and add some content.
		await headerSwitch[ 0 ].click();
		await footerSwitch[ 0 ].click();

		await page.click( 'thead .wp-block-table__cell-content' );
		await page.keyboard.type( 'header' );

		await page.click( 'tbody .wp-block-table__cell-content' );
		await page.keyboard.type( 'body' );

		await page.click( 'tfoot .wp-block-table__cell-content' );
		await page.keyboard.type( 'footer' );

		// Expect the table to have a header, body and footer with written content.
		expect( await getEditedPostContent() ).toMatchSnapshot();

		// Toggle off the switches
		await headerSwitch[ 0 ].click();
		await footerSwitch[ 0 ].click();

		// Expect the table to have only a body with written content.
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
