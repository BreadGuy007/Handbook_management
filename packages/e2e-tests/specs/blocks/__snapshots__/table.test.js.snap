// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table allows header and footer rows to be switched on and off 1`] = `
"<!-- wp:table -->
<table class=\\"wp-block-table\\"><thead><tr><th>header</th><th></th></tr></thead><tbody><tr><td>body</td><td></td></tr><tr><td></td><td></td></tr></tbody><tfoot><tr><td>footer</td><td></td></tr></tfoot></table>
<!-- /wp:table -->"
`;

exports[`Table allows header and footer rows to be switched on and off 2`] = `
"<!-- wp:table -->
<table class=\\"wp-block-table\\"><tbody><tr><td>body</td><td></td></tr><tr><td></td><td></td></tr></tbody></table>
<!-- /wp:table -->"
`;

exports[`Table allows text to by typed into cells 1`] = `
"<!-- wp:table -->
<table class=\\"wp-block-table\\"><tbody><tr><td>This</td><td>is</td></tr><tr><td>table</td><td>block</td></tr></tbody></table>
<!-- /wp:table -->"
`;

exports[`Table displays a form for choosing the row and column count of the table 1`] = `
"<!-- wp:table -->
<table class=\\"wp-block-table\\"><tbody><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr><tr><td></td><td></td><td></td><td></td><td></td></tr></tbody></table>
<!-- /wp:table -->"
`;
